datalad/tests/test_annexrepo.py

```diff
@@ -0,0 +1,36 @@
+# emacs: -*- mode: python-mode; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Tests to test implementation of class AnnexRepo
+
+Note: There's not a lot to test by now.
+
+"""
+__author__ = 'Benjamin Poldrack'
+
+import os.path
+from shutil import rmtree
+
+from nose.tools import assert_raises, assert_is_instance, assert_true
+from git.exc import GitCommandError
+
+from datalad.support.annexrepo import AnnexRepo
+from datalad.tests.utils import with_tempfile, with_testrepos, assert_cwd_unchanged
+
+
+@assert_cwd_unchanged
+@with_testrepos(flavors=['local'])
+@with_tempfile
+def test_AnnexRepo(src, dst):
+
+    ar = AnnexRepo(dst, src)
+    assert_is_instance(ar, AnnexRepo, "AnnexRepo was not created.")
+    assert_true(os.path.exists(os.path.join(dst, '.git', 'annex')))
+
+    #do it again should raise GitCommandError since git will notice there's already a git-repo at that path
+    assert_raises(GitCommandError, AnnexRepo, dst, src)
```

Comment:
ok,.... I think logic is not right here.... AnnexRepo class probably should just take "dst" as the argument, and then have a method "clone" which would get it actually installed if not there yet or puke if there already.  If annex repo exists at 'dst' it should be ok to create AnnexRepo instance again.  the same goes for the Dataset.
You are welcome to add default "src=None" argument to ```__init__``` and if not none -- run "clone".

Alternative to the former, we might use chaining pattern.  Michael opposed something like that in PyMVPA but I think it is quite neat and other toolkits use it as well
```
class C(object):
    def __init__(self, d):
        self.d = d

    def clone(self, src):
        print("cloning %s into %s" % (src, self.d))
        return self

    def get(self, f):
        print("Getting file %s" % f)
        return self

C('/tmp/test').clone('origin').get('data.dat')
```